Subject: Allow FXO channels to send out calls even before someone calls in through them
Author: Tzafrir Cohen <tzafrir.cohen@xorcom.com>
Bug: http://issues.asterisk.org/view.php?id=14577
Last-Update: 2010-03-09

This rxisoffhook does not work. Before 1.6.0 it was wrapped in a #ifdef
DAHDI_CHECK_HOOKSTATE (or rather: ZAP_CHECK_HOOKSTATE). In 1.6 it is 
accidentally enabled by default.

On DAHDI (the kernel) this field is not properly initialized at startup. 
And thus it will only work after an incoming ring.

This patch makes the test only apply in case of CAS channels. For analog
channels we have a red alarm to signal a line with no battery.

--- a/channels/chan_dahdi.c
+++ b/channels/chan_dahdi.c
@@ -10704,8 +10704,8 @@ static inline int available(struct dahdi_pvt *p, int channelmatch, ast_group_t g
 				/* When "onhook" that means no battery on the line, and thus
 				  it is out of service..., if it's on a TDM card... If it's a channel
 				  bank, there is no telling... */
-				if (par.rxbits > -1)
-					return 1;
+				if (par.rxbits <= -1)
+					return 1; /* Not CAS: not a channel bank */
 				if (par.rxisoffhook)
 					return 1;
 				else
